src/Service/Modules/TariffModule.cs

```diff
@@ -62,10 +61,11 @@ private object AddTariff()
             var employeeUri = this.Context.CurrentUser.GetEmployeeUri();
             var resource = this.Bind<TariffResource>();
             resource.Links = new[] { new LinkResource("entered-by", employeeUri) };
-
-            var result = this.tariffService.Add(resource);
-
-            return this.Negotiate.WithModel(result);
+            var validator = new TariffValidator();
+            var results = validator.Validate(resource);
+            return validator.Validate(resource).IsValid
+                       ? this.Negotiate.WithModel(this.tariffService.Add(resource))
+                       : this.Negotiate.WithModel(results).WithStatusCode(HttpStatusCode.BadRequest);
```

Comment:
Do we need to change our error catching for this format?
Should we make a response processor for it?

Reply:
Yes, I've changed how we handle the errors but only ever so slightly.. that commit is added now so you can see. I'd say that it's fine using the model that they build out, there isn't anything on the client that requires major changes... but then maybe there is something i've missed?

Reply:
No I think you're right